src/cpp/session/http/SessionHttpConnectionUtils.cpp

```diff
@@ -210,6 +222,62 @@ bool checkForSuspend(boost::shared_ptr<HttpConnection> ptrConnection)
       return false;
    }
 }
+
+bool checkForInterrupt(boost::shared_ptr<HttpConnection> ptrConnection)
+{
+   using namespace rstudio::core::json;
+   if (isMethod(ptrConnection, "interrupt"))
+   {
+      JsonRpcRequest jsonRpcRequest;
+      core::Error error = parseJsonRpcRequest(
+               ptrConnection->request().body(),
+               &jsonRpcRequest);
+      if (error)
+      {
+         ptrConnection->sendJsonRpcError(error);
+      }
+      else
+      {
+         using namespace core::system;
+         
+         // check to see if R is running a child process -- if so, forward
+         // the interrupt to that process; otherwise, just let R interrupt
+         // next time the event loop gets focus.
+         //
+         // TODO: should we be more forceful (SIGTERM?) to ensure that
```

Comment:
I don't think we should promote a `SIGINT` to a `SIGTERM`.

Reply:
If terminals are subprocesses then I think this will over-aggressively interrupt terminals (I can't believe that interrupting R means interrupt all the running terminals).

I think what we ideally want is to be able to interrupt "system" based child processes only. I don't know if there is a straightforward way to do that.